server/app/controllers/TaskSolver.scala

```diff
@@ -39,26 +44,54 @@ class TaskSolver @Inject()(executor: RuntimeSuiteExecutor, dao: Dao, val message
 
     val task = TryFuture(dao.getTask(new Date(year), TaskType.withName(taskType), timeuuid))
     task.map {
-      case Some(t) => Ok(views.html.task(t.description, solutionForm.fill(SolutionForm(t.solutionTemplate))))
+      case Some(t) => Ok(views.html.task(t.description, solutionForm.fill(SolutionForm(t.solutionTemplate, year, taskType, timeuuid.toString))))
       case None => notFound
     }.recover { case NonFatal(e) => notFound }
   }
 
-  def taskStream = WebSocket.acceptWithActor[String, JsValue] { req => out =>
-    SimpleWebSocketActor.props(out, (solution: String) =>
-      ObservableRunner(executor(
-        Class.forName("tasktest.SubArrayWithMaxSumTest").asInstanceOf[Class[Suite]],
-        Class.forName("tasktest.SubArrayWithMaxSumSolution").asInstanceOf[Class[AnyRef]],
-        solution)).map(Line(_)),
-      Some(Line("Compiling...")))
+  def taskStream = WebSocket.acceptWithActor[JsValue, JsValue] { req => out =>
+    def error(msg: String = "") = Observable.error(new SolverException(s"Cannot run $msg"))
+
+    SimpleWebSocketActor.props(out, (fromClient: JsValue) => try {
+        val solution = (fromClient \ "solution").as[String]
+        val year = (fromClient \ "year").as[Long]
+        val taskType = (fromClient \ "taskType").as[String]
+        val timeuuid = (fromClient \ "timeuuid").as[String]
+        dao.getTask(new Date(year), TaskType.withName(taskType), UUID.fromString(timeuuid)).map {
+          case Some(t) => ObservableRunner(executor(solution, t.suite)).map(Line(_))
+          case _ => error()
+        }
+      } catch {
+        case NonFatal(e) => Future.successful(error(e.getMessage))
+      },
+      Some(Line("Compiling..."))
+    )
+  }
+
+  def runtimeTaskStream = WebSocket.acceptWithActor[JsValue, JsValue] { req => out =>
+    SimpleWebSocketActor.props(out, (fromClient: JsValue) => Future.successful((try {
+        val suiteClass = "tasktest.SubArrayWithMaxSumTest" // (fromClient \ "suiteClass").as[String]
```

Comment:
but how it is working, it a kind of magic? xD
`executor: RuntimeSuiteExecutor with DynamicSuiteExecutor`